### ISSUE TITLE
Support automatically adding to PATH on macOS

### ISSUE THREAD

**Issue by DanTup** [Dart-Code/Dart-Code#4612]
Currently the work done for https://github.com/Dart-Code/Dart-Code/issues/4604 is Windows-only.

We need to evaluate options for adding to PATH on macOS. The original plan was to add symlinks to something like `/usr/bin`, however since the SDK is likely in a user-specific folder, this could do bad things for systems used by more than one user.

We should see if there's a reasonable way we could automate adding to shell profile scripts (is it save to always append? can we tell which profile script(s) are required?).

@polina-c

**Comment by DanTup** [comment]
Let's do what's written on https://docs.flutter.dev/get-started/install/macos/mobile-ios#add-flutter-to-your-path, and just assume zsh and append it there.

It's possible that doing so here won't apply to `PATH` for apps not launched from inside the terminal (like launching VS Code from the dock), but since we should also be writing the VS Code setting, then (unless the user deletes the setting) it should still be fine.

(cc @anderdobo)

**Comment by sethladd** [comment]
how about this:

`echo $SHELL`

if bash, append `export PATH=$PATH:$FLUTTER_ROOT/bin` to `~/.bash_profile`

if zsh , do the zshell thing.

**Comment by DanTup** [comment]
Good idea! I didn't know about `SHELL` - but assuming it's set system-wide (and not just by the shell that is running) we should be able to just read it in VS Code (we have direct access to the env variables without needing to spawn/echo) and then write the export to the appropriate file.

**Comment by sethladd** [comment]
Sure, if you can read in `$SHELL` without calling `echo`, great!

either way, sounds like you found a way to detect the shell so you know what file to update.

BTW, if the user has a shell that is NOT bash or zsh, you should pop up a message like "Unable to auto-set Flutter's tools into your path, because your SHELL was detected to be FOO , and we only support setting for Bash and zsh. To manually set Flutter into your Path, add actual/path/to/flutter/bin to your shell's path"

Of course, you can come up with more terse wording.

**Comment by DanTup** [comment]
> BTW, if the user has a shell that is NOT bash or zsh, you should pop up a message like

We could also not show the option if `$SHELL` isn't one of those, and retain the current behaviour (eg. just show the "Open Instructions" button that links to the Flutter website, which could be expanded). It seems pointless to show the option and let you click it if we know we wouldn't be able to action it.

(probably it's a bit of an edge case though, I suspect the number of users that aren't using one of those shells on macOS is probably very small)

**Comment by sethladd** [comment]
yes, if $SHELL isn't bash or zsh, then show a helpful message and direct the user to read the instructions on the Flutter website to set the path. Good idea!

btw, when do you think you can have this complete? we're trying to decide if we need to add a workaround in a thing we're doing. We won't do the workaround if this is coming in the next 2-3 days. Thanks for considerin!

**Comment by DanTup** [comment]
@sethladd I should be able to look at this today, although the next scheduled release isn't until the end of the month.

It's a small enough fix that it should be safe to cherry-pick into a patch release though.

I will note that this shouldn't be necessary if you're using the built-in shell inside VS Code though - we already automatically add to `PATH` for built-in terminals, this functionality is only for external terminals outside of VS Code.

**Comment by DanTup** [comment]
@sethladd @anderdobo I've published a pre-release of the extension today that includes this work (as well as another bug I found while testing - https://github.com/Dart-Code/Dart-Code/issues/5709). To test it, ensure you're on the pre-release and that you have `v3.119.20250918` (and have restarted the extensions after updating).

<img width="703" alt="Dart pre-release versions" src="https://user-images.githubusercontent.com/1078012/218796750-90a70b0b-4957-468c-8eb4-688095cde112.png">

Assuming it works as you'd like, let me know if the pre-release is good enough for your use, or if you think we should cherry-pick this into a stable release (I could do this tomorrow).

Note: This change was macOS-only, however it occurs to me that the changes as-is would probably work for Linux too (for those two shells that we support),. I've opened https://github.com/Dart-Code/Dart-Code/issues/5711 for this, which I'll just include in the next stable release unless it's important to you to also include now in a pre-release or a stable patch.

**Comment by sethladd** [comment]
Thanks so much for the quick turnaround! I'll test now.

**Comment by sethladd** [comment]
Seemed to work great, thank you!! Tested on a Mac with VS Code

<img width="759" height="165" alt="Image" src="https://github.com/user-attachments/assets/2a1d781d-39c8-40a7-842d-27103fa81e9f" />
<img width="684" height="199" alt="Image" src="https://github.com/user-attachments/assets/37c807b5-7ea4-4a0b-bb7b-c7e584f08a75" />

And it put the following into my `~/.bash_profile`: `export PATH="/Users/sethladd/flutter/bin:$PATH"`

**Comment by sethladd** [comment]
> you think we should cherry-pick this into a stable release (I could do this tomorrow).

I personally would like a new stable release, as it simplifies things for us. However, I defer to you and Ander to assess the risk. If this is the only change , then I think the risk is low and you should make a new stable release.

**Comment by sethladd** [comment]
> I will note that this shouldn't be necessary if you're using the built-in shell inside VS Code though - we already automatically add to PATH for built-in terminals, this functionality is only for external terminals outside of VS Code.

Ack

> I've opened #5711 for this, which I'll just include in the next stable release unless it's important to you to also include now in a pre-release or a stable patch.

Personally, we don't need #5711 to be included in a stable path release.

**Comment by DanTup** [comment]
@sethladd thanks for testing! I'll push a stable build with this tomorrow (it will be just this and https://github.com/Dart-Code/Dart-Code/issues/5709 cherry-picked, both of which are low risk IMO).

**Comment by sethladd** [comment]
Sounds awesome, thanks!

**Comment by DanTup** [comment]
@sethladd this has been published in the stable v3.118.2 of the Dart extension, so the fix is now live to both stable and pre-release users. Users will need to let the extension update and then click the "Restart Extensions" button or restart VS Code after installing.

**Comment by sethladd** [comment]
Thank you so much!
